api/src/main/java/jakarta/data/repository/Update.java

```diff
@@ -35,9 +35,23 @@
  *     <li>An array of entities to be updated.</li>
  * </ul>
  * <p>The return type of the annotated method must be {@code void}, {@code boolean}, a numeric primitive type
- * (such as {@code int}), or a corresponding primitive wrapper type (such as {@link Integer}).
+ * (such as {@code int}), a corresponding primitive wrapper type (such as {@link Integer}), or the same type as the parameter.
+ * </p>
+ * <p>
+ * The return type of an annotated method that accepts an {@code Iterable} or array of entities as the parameter must
+ * have a return type that is {@code void}, {@code Void}, or an {@code Iterable} or array of the entity.
  * A boolean return type indicates whether a matching entity was found in the database to update.
  * A numeric return type indicates how many matching entities were found in the database to update.
+ * The return type of an annotated method that accepts an {@code Iterable} or array of entities as the parameter must
+ * have a return type that is {@code void}, {@code Void}, or an {@code Iterable} or array of the entity.
```

Comment:
The same text is duplicated in two places here, probably a copy/paste error from the `@Insert` method given that it leaves out the numeric and boolean types. The correct return type requirements were just stated prior to this so we can just remove this.
We do need to say something about what happens with the return value when some of the entities are not found. It will make sense to include that here,
```suggestion
 * A boolean return type indicates whether a matching entity was found in the database to update.
 * A numeric return type indicates how many matching entities were found in the database to update.
 * An entity return type indicates the updated entity if found in the database. If the entity is not found
 * in the database or has a non-matching version, then {@code null} is returned.
 * An {@code Iterable} or array return type includes all matching entities that are found in the database,
 * skipping over entities that are not present in the database or have a non-matching version.
```